src/openalea/lpy/lsysparameters.py

```diff
@@ -0,0 +1,652 @@
+        return self.get_available_scalar_types()+self.get_available_graphical_types()
+
+    def get_available_scalar_types(self):
+        return scalartypemap.values()
+
+    def get_graphicalparameter_managers(self):
+        from openalea.lpy.gui.objectmanagers import get_managers
+        return get_managers()
+
+    def get_available_graphical_types(self):
+        return list(self.get_graphicalparameter_managers().keys())
+
+    def add(self, name, value, ptype = None, category = None, **params):
+        if ptype is None:
+            ptype = type(value)
+            ptype = scalartypemap.get(ptype,graphictypemap.get(ptype,ptype))
+        if ptype in self.get_available_scalar_types():
+            self.add_scalar(name, value, ptype, category, **params)
+        elif ptype in self.get_available_graphical_types():
+            self.add_graphicalparameter(name, value, ptype, category, **params)
+        else:
+            raise TypeError(ptype)
+
+    def add_category(self, name, **params):
+        category = Category({ 'name' : category, 'enabled' : True })
+        category.update(params)
+        self.categories[name] = category
+
+    def set_defaut_category(self, name):
+        self.default_category_name = name
+
+    def add_scalar(self, name, value, ptype = None, category = None, **params):
+        from .parameters.scalar import ProduceScalar
+        if ptype is None:
+            ptype = scalartypemap[type(value)]
+        if category is None:
+            category = self.default_category_name
+        assert ptype in self.get_available_scalar_types()
+        scalar = ProduceScalar((name, ptype, value))
+        scalar.__dict__.update(**params)
+        scalar.category = category
+
+        self._add_scalar(category, scalar)
+
+    def _add_scalar(self, category, scalar):
+        categoryobj = self.get_category(category)
+        categoryobj.scalars.append(scalar)
+
+    def add_function(self, name, value, category = None):
+        self.add_graphicalparameter(name, value, 'Function', category)
+
+    def add_graphicalparameter(self, name, value, ptype = None, category = None):
+        assert ptype in self.get_available_graphical_types()
+        if ptype is None:
+            ptype = graphictypemap[type(value)]
+        if category is None:
+            category = self.default_category_name
+
+        manager = self.get_graphicalparameter_managers()[ptype]
+        manager.setName(value, name)
+
+        self._add_graphicalparameter(category, manager, value)
+
+    def _add_graphicalparameter(self, category, manager, value):
+        categoryobj = self.get_category(category)
+        categoryobj.items.append((manager, value))
+
+    def set_option(self, name, value):
+        self.execOptions[name] = value
+
+    def get_option(self, name):
+        self.execOptions[name]
+
+    def get_options(self):
+        return self.execOptions
+
+    def set_color(self, index, value, name = None):
```

Comment:
It could be made also with get_colors which gives a dict that could be manipulated as you want.
unset_color is added in 280e4c3adde6f0c9f6a0e92e5cf5a54a57048059